server/manager/src/main/java/org/apache/accumulo/manager/tableOps/namespace/delete/DeleteNamespace.java

```diff
@@ -41,7 +49,26 @@ public long isReady(long id, Manager environment) throws Exception {
   }
 
   @Override
-  public Repo<Manager> call(long tid, Manager environment) {
+  public Repo<Manager> call(long tid, Manager environment)
+      throws AcceptableThriftTableOperationException {
+    try {
+      // Namespaces.getTableIds(..) uses the following cache, clear the cache to force
+      // Namespaces.getTableIds(..) to read from zookeeper.
+      environment.getContext().clearTableListCache();
+      // Since we have a write lock on the namespace id and all fate table operations get a read
+      // lock on the namespace id there is no need to worry about a fate operation concurrently
+      // changing table ids in this namespace.
+      List<TableId> tableIdsInNamespace =
+          Namespaces.getTableIds(environment.getContext(), namespaceId);
+      if (!tableIdsInNamespace.isEmpty()) {
+        throw new AcceptableThriftTableOperationException(null, null, TableOperation.DELETE,
+            TableOperationExceptionType.OTHER,
```

Comment:
Could we add a different TableOperationExceptionType instead of relying on OTHER?